### PR TITLE
style: commitlint

### PR DESCRIPTION
**Pull request by tristanhcole**
**Describe the Pull Request**

Starting to setup commitlint for cleaner commits, automated release notes and to follow conventionalcommits.org. This is a precursor to automated SemVer. Also updated contributing guidelines.

PyCharm users this is a good plugin: https://plugins.jetbrains.com/plugin/9861-git-commit-template

**Todo**

- [x] Link relevant [trello card](https://trello.com/b/PA2LhlOh/sempo-dev) or [related issue](https://github.com/teamsempo/SempoBlockchain/issues) to the pull request
- [x] Request review from relevant @person or team
- [x] QA your pull request. This includes running the app, login, testing changes etc.
- [ ] Bump [backend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/config.py#L7) and/or [frontend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/app/package.json#L3) versions following Semver

### CODE REVIEW

**Comment by tristanhcole** [comment]
[Make Github bump version number automatically](https://trello.com/c/bfq6q3bj/272-make-github-bump-version-number-automatically)

**Comment by tristanhcole** [comment]
Interested to get your thoughts on this @enjeyw and @michielderoos :)

**Comment by tristanhcole** [comment]
I personally like this tool. Marking as ready for review for feedback and comments!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1 +1,138 @@
...
+The default graphite width of 10mm is always used for performance reasons.
+```
+
+## Get Started
+
+TODO
+
+### Lint / Code Formatter
+
+TODO
```
Comment (michielderoos) [inline CONTRIBUTING.md:134]: We should come up with a standard linter + linter guidelines soon IMO

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1 +1,138 @@
...
+
+Please read and follow our [Code of conduct](CODE_OF_CONDUCT.md).
+
+## How can I contribute?
+
+- Improve documentation
+- Give feedback on issues
+- Fix bugs and implement new features
+- [Bug report](https://github.com/teamsempo/SempoBlockchain/issues/new?assignees=&labels=&template=bug_report.md&title=)
+- [Feature request](https://github.com/teamsempo/SempoBlockchain/issues/new?assignees=&labels=&template=feature_request.md&title=)
```
Comment (michielderoos) [inline CONTRIBUTING.md:23]: Is there some integration between issues and Trello?

Comment (tristanhcole) [reply CONTRIBUTING.md:23]: I did setup a Zapier so that new github issues create a trello card, but not really tbh

File: CONTRIBUTING.md (hunk 1 above)
Comment (tristanhcole) [reply CONTRIBUTING.md:134]: agree

Comment (tristanhcole) [reply CONTRIBUTING.md:134]: @enjeyw any further thoughts on python linting?

Comment (enjeyw) [reply CONTRIBUTING.md:134]: black with 121 line limit lol!

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1 +1,138 @@
...
+
+TODO
+
+### Documentation
+
+TODO
+
+### Commit message guidelines
+
+#### Atomic commits
```
Comment (enjeyw) [inline CONTRIBUTING.md:41]: how much do we care about this given we squash merge? seems like this guideline is kinda spec bloat to me... we've never mentioned anything like this before

File: app/package.json [hunk 4]
```diff
@@ -34,6 +34,8 @@
     "@babel/plugin-proposal-class-properties": "^7.8.3",
     "@babel/preset-env": "^7.9.5",
     "@babel/preset-react": "^7.9.4",
+    "@commitlint/cli": "^8.3.5",
+    "@commitlint/config-conventional": "^8.3.4",
```
Comment (enjeyw) [inline app/package.json:38]: wait, so it's the app package that dictates merge behaviour for the entire repo? seems weird to me.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1 +1,138 @@
...
+| **build**    | Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)         |
+| **ci**       | Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs) |
+| **docs**     | Documentation only changes                                                                                  |
+| **feat**     | A new feature                                                                                               |
+| **fix**      | A bug fix                                                                                                   |
+| **perf**     | A code change that improves performance                                                                     |
+| **refactor** | A code change that neither fixes a bug nor adds a feature                                                   |
+| **style**    | Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)      |
+| **test**     | Adding missing tests or correcting existing tests                                                           |
```
Comment (enjeyw) [inline CONTRIBUTING.md:87]: these are noice

File: CONTRIBUTING.md (hunk 1 above)
Comment (michielderoos) [reply CONTRIBUTING.md:134]: Swap `    ` for `	`

File: CONTRIBUTING.md (hunk 3 above)
Comment (michielderoos) [reply CONTRIBUTING.md:41]: I kind of read this as `atomic PRs`, given the squash. But I guess most PRs are atomic by nature anyway

File: app/package.json (hunk 4 above)
Comment (michielderoos) [reply app/package.json:38]: #JustNodeThings 😉

File: CONTRIBUTING.md (hunk 3 above)
Comment (tristanhcole) [reply CONTRIBUTING.md:41]: Yeah, I view it as atomic commits on master branch (so PRs)

File: app/package.json (hunk 4 above)
Comment (tristanhcole) [reply app/package.json:38]: lol we can look at [lerna](https://github.com/lerna/lerna) if we expand to multiple JS projects in this repo

**Comment by tristanhcole** [comment]
@michielderoos @enjeyw One thing I think could be improved about this PR, is also setting up a CI line that enforces commit linting, as well as checking the PR title

File: app/package.json (hunk 4 above)
Comment (enjeyw) [reply app/package.json:38]: thoughts on just shifting commit lint into a repo level package.json, since it's not really appropriate here?

**Comment by tristanhcole** [comment]
Outdated

**Pull request closed without merging**